apollo-compiler/src/test/graphql/com/example/inline_fragment_for_non_optional_field/TestQuery.java

```diff
@@ -177,6 +177,23 @@ public Hero read(ResponseReader reader) {
 
     ResponseFieldMarshaller marshaller();
 
+    default <T> T visit(Visitor<T> visitor) {
+      if (this instanceof AsHuman) {
+        return visitor.visit((AsHuman) this);
+      } else if (this instanceof AsCharacter) {
+        return visitor.visit((AsCharacter) this);
+      }
+      return visitor.visitDefault(this);
+    }
```

Comment:
So the default implementation of function in interface will lock Apollo to be used only with Java8.
Should we guard this feature by gradle plugin configuration flag then?

Reply:
Yeah. My thinking here was that most people on Android are already on Java 1.8 compatibility. But yeah, that's a big assumption. I'm gonna add an option in the ApolloExtension to option into generating visitors and so only those who opt into it needs to add the source/target compatibility. I'll be sure to leave a note in the readme as well.